hail/python/hail/matrixtable.py

```diff
@@ -2839,9 +3019,16 @@ def _repr_html_(self):
         handler=nullable(anyfunc),
     )
     def show(
-        self, n_rows=None, n_cols=None, include_row_fields=False, width=None, truncate=None, types=True, handler=None
+        self,
+        n_rows: Optional[int] = None,
+        n_cols: Optional[int] = None,
+        include_row_fields: bool = False,
+        width: Optional[int] = None,
+        truncate: Optional[int] = None,
+        types: bool = True,
+        handler: Optional[Callable[[str], Any]] = None,
```

Comment:
I'm not actually sure how to type this correctly. Even PyRight can't seem to figure it out:
```
(base) dking@wm28c-761 hail % cat bar.py    
from IPython.display import display
reveal_type(display)

(base) dking@wm28c-761 hail % pyright bar.py
WARNING: there is a new pyright version available (v1.1.349 -> v1.1.350).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

/Users/dking/projects/hail/bar.py
  /Users/dking/projects/hail/bar.py:2:13 - information: Type of "display" is "(*objs: Unknown, include: Unknown | None = None, exclude: Unknown | None = None, metadata: Unknown | None = None, transient: Unknown | None = None, display_id: Unknown | None = None, raw: bool = False, clear: bool = False, **kwargs: Unknown) -> (DisplayHandle | None)"
```

Reply:
Okay, in the next set of changes, I make the type `Optional[Callable]`.

```
from typing import Callable
from IPython.display import display
type(display)
Out[4]: function
isinstance(display, Callable)
Out[5]: True
```